.github/workflows/build_wheels.yml

```diff
@@ -0,0 +1,157 @@
+          path: cache/hdf5
+
+      # Windows HDF5
+      - uses: nuget/setup-nuget@v2
+        if: runner.os == 'Windows'
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+        if: runner.os == 'Windows'
+      - run: bash ./ci/cibw_before_all_windows.sh "${{ github.workspace }}"
+        if: runner.os == 'Windows'
+
+      # macOS HDF5
+      - run: bash ./ci/cibw_before_all_macos.sh "${{ github.workspace }}"
+        if: runner.os == 'macOS'
+      # Hack for 3.8 bug (https://github.com/pypa/cibuildwheel/pull/1871#issuecomment-2161613619)
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.8
+        if: runner.os == 'macOS' && matrix.arch == 'arm64'
+
+      # Linux emulation for aarch64 support
+      # https://cibuildwheel.pypa.io/en/stable/faq/#emulation
+      - uses: docker/setup-qemu-action@v3
+        with:
+          platforms: all
+        if: runner.os == 'Linux' && matrix.arch == 'aarch64'
+
+      # Triage build, and set outputs.skip = 1 if on aarch64 and we don't want to build the wheel
+      - run: bash ./ci/triage_build.sh "${{ matrix.arch }}" "${{ github.event.pull_request.head.sha || github.sha }}" "${{ matrix.python }}"
+        id: triage
+
+      # Now actually build the wheels
+      - uses: pypa/cibuildwheel@v2.19.0
+        env:
+          # Note that tool.cibuildwheel "skip" in pyproject.toml also limits what gets built
+          CIBW_ARCHS: ${{ matrix.arch }}
+          CIBW_BUILD: ${{ env.CIBW_BUILD }}
+          CIBW_SKIP: ${{ env.CIBW_SKIP }}
+          CIBW_BEFORE_BUILD: ${{ env.CIBW_BEFORE_BUILD }}
+          CIBW_BUILD_FRONTEND: ${{ env.CIBW_BUILD_FRONTEND }}
+          TOX_TEST_LIMITED: ${{ env.TOX_TEST_LIMITED }}
+        if: steps.triage.outputs.skip != '1'
+
+      # And upload the results
+      - uses: actions/upload-artifact@v4
+        with:
+          name: cibw-wheels-${{ matrix.os }}-${{ strategy.job-index }}
+          path: ./wheelhouse/*.whl
+        if: steps.triage.outputs.skip != '1'
+
+  upload_wheels:
+    name: Upload wheels
+    needs: build_wheels
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: cibw-wheels-*
+          merge-multiple: true
+          path: dist
+      - run: pwd && ls -alt . && ls -alt dist/
+      # If it's a PR, do nothing else.
+      # If it's a scheduled run, it was built with nightly NumPy, push to scientific-python:
+      - name: Upload wheels to Anaconda Cloud as nightlies
+        uses: scientific-python/upload-nightly-action@b67d7fcc0396e1128a474d1ab2b48aa94680f9fc # 0.5.0
+        with:
+          artifacts_path: dist
+          anaconda_nightly_upload_token: ${{ secrets.ANACONDA_ORG_UPLOAD_TOKEN }}
+        if: github.repository_owner == 'h5py' && github.event_name == 'schedule'
+      # If it's a push to master, create a draft release;
```

Comment:
[Here](https://github.com/openmeeg/openmeeg/releases) is an example of what it looks like. I don't think it creates much noise because you can just have a `dev` release that gets updated.

At a minimum I would suggest keeping it *temporarily* at least, just to make sure it can actually create a release once this PR lands in `main`. If it fails then you get that failure now and can debug immediately rather than when you actually try to cut a release and it fails. That's an annoying situation!

Reply:
To save some clicks, the `main` `dev` "release" shows up as a draft in the list of releases:

![Screenshot from 2024-06-13 11-43-24](https://github.com/h5py/h5py/assets/2365790/5237a7ae-50b0-4767-8e43-539e57a91384)

And it *doesn't* show up as the latest release on the GitHub landing page, which is nice:

![Screenshot from 2024-06-13 11-43-35](https://github.com/h5py/h5py/assets/2365790/933d2075-fd01-4238-9f30-29c765e1b903)

Again, might not make sense as a permanent thing but at least temporarily it's fairly unobtrusive and would allow you to make sure wheels can get added as assets to a GitHub release, increasing the likelihood that it'll do the right thing when you do eventually cut a release.